aks-node-controller/app.go

```diff
@@ -116,16 +119,35 @@ func (a *App) Provision(ctx context.Context, flags ProvisionFlags) error {
 	if cmd.ProcessState != nil {
 		exitCode = cmd.ProcessState.ExitCode()
 	}
-	// Is it ok to log a single line? Is it too much?
 	slog.Info("CSE finished", "exitCode", exitCode, "stdout", stdoutBuf.String(), "stderr", stderrBuf.String(), "error", err)
 	return err
 }
 
+// notifyProvisionFailure creates the provision.complete sentinel file if a provisioning
+// error occurred before the normal provisioning scripts had a chance to write it.
+// This prevents the provision-wait mode from blocking until timeout when we fail fast
+// (e.g. due to invalid or unsupported configuration).
+func (a *App) notifyProvisionFailure(runErr *error) {
+	if runErr == nil || *runErr == nil { // success path or nil pointer
+		return
+	}
+	// Avoid clobbering an existing file if provisioning scripts actually created it before the error surfaced.
+	if _, statErr := os.Stat(provisionCompleteFilePath); statErr == nil {
+		return // file already exists
+	} else if !errors.Is(statErr, os.ErrNotExist) { // unexpected stat error
+		slog.Error("failed to stat provision.complete file", "error", statErr)
+		return
+	}
+	if writeErr := os.WriteFile(provisionCompleteFilePath, []byte{}, 0600); writeErr != nil {
+		slog.Error("failed to write provision.complete file", "error", writeErr)
```

Comment:
I would probably also log `provisionCompleteFilePath`

Reply:
Oh it's an empty file. Just an indicator for provision complete

Reply:
When we can't write to the file it can be useful for the user to know what file we're trying to write

Reply:
oh you mean log the filepath. I thought you were saying the content. Updated.